File: frontend/src/types/Appointment.ts
import type { Treatment } from "./Treatment";

export interface Appointment {
  id: number;
  dateTime: string;
  dentistName: string;
  treatment: Treatment | null;
  durationMinutes: number;
}

export interface CreateAppointmentDto {
  patientId: number;
  dateTime: string;
  dentistName: string;
  treatment: Treatment | null;
}

File: backend/Pms.Api/Models/Appointment.cs
namespace Pms.Api.Models
{
    public class Appointment
    {
        public int Id { get; set; }
        public DateTime DateTime { get; set; }
        public string DentistName { get; set; } = string.Empty;
        public Treatment? Treatment { get; set; }
        public int DurationMinutes { get; set; }
        public int PatientId { get; set; }
        public Patient Patient { get; set; } = null!;
        }
}